pkg/idp/validation_test.go

```diff
@@ -119,6 +119,46 @@ func TestValidate(t *testing.T) {
 			expectedResult: nil,
 			expectedError:  nil,
 		},
+		{
+			name:     "CreateIdPSuccessWithOIDCDiscovery",
+			method:   http.MethodPost,
+			endpoint: "",
+			body: func() []byte {
+				conf := new(Configuration)
+				conf.ID = "google_generic"
+				conf.Provider = "generic"
+				conf.IssuerURL = "mock-url"
+				conf.SubjectSource = "me"
+				conf.Scope = []string{}
+				conf.Mapper = "mock-url"
+
+				marshal, _ := json.Marshal(conf)
+				return marshal
+			},
+			expectedResult: nil,
+			expectedError:  nil,
+		},
+		{
+			name:     "CreateIdPSuccessWithoutOIDCDiscovery",
+			method:   http.MethodPost,
+			endpoint: "",
+			body: func() []byte {
+				conf := new(Configuration)
+				conf.ID = "google_generic"
+				conf.Provider = "generic"
+				conf.IssuerURL = "mock-url"
```

Comment:
`IssuerURL` should be empty

Reply:
I removed the requirement on the IssuerURL so it can actually be empty in case authurl and tokenurl are present, for the generic provider. I'm adjusting the test right now.